test/cpp/api/serialize.cpp

```diff
@@ -668,96 +669,64 @@ TEST(SerializeTest, Optim_RMSprop) {
-  torch::save(model1, model_tempfile.name);
-  torch::load(model2, model_tempfile.name);
-  torch::load(model3, model_tempfile.name);
-
-  auto param1 = model1->named_parameters();
-  auto param2 = model2->named_parameters();
-  auto param3 = model3->named_parameters();
-  for (const auto& p : param1) {
-    ASSERT_TRUE(p->allclose(param2[p.key()]));
-    ASSERT_TRUE(param2[p.key()].allclose(param3[p.key()]));
-  }
-  // Make some optimizers
-  auto optim1 = LBFGS(
-      {torch::optim::OptimizerParamGroup(model1->parameters())}, options);
-  auto optim2 = LBFGS(
-      model2->parameters(), options);
-  auto optim2_2 = LBFGS(
-      model2->parameters(), options);
-  auto optim3 = LBFGS(
-      model3->parameters(), options);
-  auto optim3_2 = LBFGS(
-      model3->parameters(), options);
+  auto model1_params = model1->parameters();
+  // added a tensor for lazy init check - when all params do not have entry in buffers
+  model1_params.emplace_back(torch::randn({2,3}));
+  auto optim1 = torch::optim::LBFGS(model1_params, torch::optim::LBFGSOptions());
 
   auto x = torch::ones({10, 5});
-
-  auto step = [&x](torch::optim::LossClosureOptimizer& optimizer, Linear model) {
+  auto step = [&x](torch::optim::Optimizer& optimizer, Linear model) {
     optimizer.zero_grad();
     auto y = model->forward(x).sum();
     y.backward();
     auto closure = []() { return torch::tensor({10}); };
     optimizer.step(closure);
   };
 
-  // Do 2 steps of model1
-  step(optim1, model1);
   step(optim1, model1);
 
-  // Do 2 steps of model 2 without saving the optimizer
-  step(optim2, model2);
-  step(optim2_2, model2);
+  at::Tensor d, t, H_diag, prev_flat_grad, prev_loss;
+  std::deque<at::Tensor> old_dirs, old_stps;
 
-  // Do 1 step of model 3
-  step(optim3, model3);
-
-  // save the optimizer
-  auto optim_tempfile = c10::make_tempfile();
-  torch::save(optim3, optim_tempfile.name);
-  torch::load(optim3_2, optim_tempfile.name);
+  const auto& params_ = optim1.param_groups()[0].params();
+  auto key_ = c10::guts::to_string(params_[0].unsafeGetTensorImpl());
+  const auto& optim1_state = static_cast<const LBFGSParamState&>(*(optim1.state().at(key_).get()));
+  d = optim1_state.d();
+  t = at::tensor(optim1_state.t());
+  H_diag = optim1_state.H_diag();
+  prev_flat_grad = optim1_state.prev_flat_grad();
+  prev_loss = at::tensor(optim1_state.prev_loss());
+  old_dirs = optim1_state.old_dirs();
 
-  auto& optim3_2_param_groups = optim3_2.param_groups();
-  auto& optim3_param_groups = optim3.param_groups();
-  auto& optim3_2_state = optim3_2.state();
-  auto& optim3_state = optim3.state();
+  // write buffers to the file
+  auto optim_tempfile_old_format = c10::make_tempfile();
+  torch::serialize::OutputArchive output_archive;
+  output_archive.write("d", d);
+  output_archive.write("t", t);
+  output_archive.write("H_diag", H_diag);
+  output_archive.write("prev_flat_grad", prev_flat_grad);
+  output_archive.write("prev_loss", prev_loss);
```

Comment:
maybe pass `/*is_buffer=*/true` for the above five function calls, to be consistent with old version code